database/sqlserver/sqlserver.go

```diff
@@ -117,16 +121,43 @@ func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
 	return ss, nil
 }
 
-// Open a connection to the database
+// Open a connection to the database.
 func (ss *SQLServer) Open(url string) (database.Driver, error) {
 	purl, err := nurl.Parse(url)
 	if err != nil {
 		return nil, err
 	}
 
-	db, err := sql.Open("sqlserver", migrate.FilterCustomQuery(purl).String())
-	if err != nil {
-		return nil, err
+	useMsiParam := purl.Query().Get("useMsi")
+	useMsi := false
+	if len(useMsiParam) > 0 {
+		useMsi, err = strconv.ParseBool(useMsiParam)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var db *sql.DB
+	if useMsi {
```

Comment:
How would we handle the case when both `useMsi` and a password are specified?

Reply:
In this case I think it makes sense to default to MSI auth and could then also log a message to the console saying something along the lines of "Both MSI Auth and Password Auth were requested, defaulting to MSI Auth. Password will be ignored" (can update the README with this behavior as well). Any sort of fallback to password auth should probably be handled by the caller.

Reply:
Any objections to making these options mutually exclusive? e.g. error if both are specified?

Reply:
That sounds good to me, made the change.